e2e/cypress/integration/enterprise/saml/okta_login_spec.js

```diff
@@ -0,0 +1,272 @@
+
+    const regular1 = users.regulars['samluser-1'];
+    const guest1 = users.guests['samlguest-1'];
+    const guest2 = users.guests['samlguest-2'];
+    const admin1 = users.admins['samladmin-1'];
+    const admin2 = users.admins['samladmin-2'];
+
+    const newConfig = {
+        ServiceSettings: {
+            SiteURL: Cypress.config('baseUrl'),
+        },
+        SamlSettings: {
+            Enable: true,
+            EnableSyncWithLdap: false,
+            EnableSyncWithLdapIncludeAuth: false,
+            Verify: true,
+            Encrypt: true,
+            SignRequest: true,
+            IdpUrl: Cypress.env('oktaBaseUrl') + '/app/' + Cypress.env('oktaMMAppName') + '/' + Cypress.env('oktaMMEntityId') + '/sso/saml',
+            IdpDescriptorUrl: 'http://www.okta.com/' + Cypress.env('oktaMMEntityId'),
+            IdpMetadataUrl: Cypress.env('oktaBaseUrl') + '/app/' + Cypress.env('oktaMMEntityId') + '/sso/saml/metadata',
+            AssertionConsumerServiceURL: Cypress.config('baseUrl') + '/login/sso/saml',
+            SignatureAlgorithm: 'RSAwithSHA1',
+            CanonicalAlgorithm: 'Canonical1.0',
+            IdpCertificateFile: 'saml-idp.crt',
+            PublicCertificateFile: 'saml-public.crt',
+            PrivateKeyFile: 'saml-private.key',
+            IdAttribute: '',
+            GuestAttribute: '',
+            EnableAdminAttribute: false,
+            AdminAttribute: '',
+            FirstNameAttribute: '',
+            LastNameAttribute: '',
+            EmailAttribute: 'Email',
+            UsernameAttribute: 'Username',
+            LoginButtonText: 'SAML',
+        },
+        ExperimentalSettings: {
+            UseNewSAMLLibrary: true
+        },
+        GuestAccountsSettings: {
+            Enable: true
+        },
+    };
+
+    let testSettings;
+
+    describe('SAML Login flow', () => {
+        before(() => {
+            cy.oktaAddUsers(users);
+            cy.apiUpdateConfig(newConfig).then(() => {
+                cy.apiGetConfig().then((response) => {
+                    cy.setTestSettings(loginButtonText, response.body).then((_response) => {
+                        testSettings = _response;
+                    });
+                });
+            });
+        });
+
+        it('Saml login new and existing MM regular user', () => {
+            testSettings.user = regular1;
+
+            cy.oktaGetOrCreateUser(testSettings.user).then((oktaUserId) => {
+                cy.oktaDeleteSession(oktaUserId);
+                cy.doSamlLogin(testSettings).then(() => {
+                    cy.doOktaLogin(testSettings.user).then(() => {
+                        cy.skipOrCreateTeam(testSettings, oktaUserId).then(() => {
+                            cy.doSamlLogout(testSettings).then(() => {
+                                cy.oktaDeleteSession(oktaUserId);
+                            });
+                        });
+                    });
+                });
+            });
```

Comment:
I assume this is the test to verify SAML login new user. In this case:
a) can you please add a comment to indicate this. (Also please do it throughout the test wherever applicable). 
b) can you please add a step to delete the user in Okta before this test.
c) we should also be deleting the user in Mattermost. (This could be done by updating the existing user's email & username).

Reply:
Regarding point (b), please ignore this. We just need to ensure that the user is new in Mattermost. Deleting the user in Okta would not make a difference. 

And regarding point (c), as discussed, we can go ahead with the assumption that the test runs on a clean setup everyday and that would ensure that the users are not present in the Mattermost instance. Can you just add this assumption as a comment in the test so that we consider this when we analyse an inconsistent failure.